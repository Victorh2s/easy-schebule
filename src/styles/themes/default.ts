export const defaultTheme = {
  Light: {
    button: '#5300FF',
    success: '#2AAA50',
    error: '#D42029',
    gray: '#A4A69F',
    'font-color': '#FFFFFF',
    'logo-color': '#05543A',
    'pendente-color': '#F73A4E',
    'fazendo-color': '#FF911C',
    'feito-color': '#2AE850',
    'background-color': '#0f0f0f',
    'link-color': '#15A89F',
    'background-form': '#171717',
    'border-input': '#D2D2D4',
  },
  Dark: {
    button: '#5300FF',
    success: '#2AAA50',
    error: '#D42029',
    gray: '#A4A69F',
    'font-color': '#0f0f0f',
    'logo-color': '#05543A',
    'pendente-color': '#F73A4E',
    'fazendo-color': '#FF911C',
    'feito-color': '#2AE850',
    'background-color': '#FFFFFF',
    'link-color': '#15A89F',
    'background-form': '#EDEDEF',
    'border-input': '#D2D2D4',
  },
} as const;
