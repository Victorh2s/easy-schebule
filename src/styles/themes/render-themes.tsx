import { render, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from './default';
import { BrowserRouter } from 'react-router-dom';

export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <BrowserRouter>
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>{' '}
    </BrowserRouter>
  );
};
