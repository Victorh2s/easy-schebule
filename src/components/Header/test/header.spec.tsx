import { describe, test, expect } from 'vitest';
import { Header } from '../index';
import { renderTheme } from '../../../styles/themes/render-themes';

describe('Header component', () => {
  test('renders correctly with user menu', () => {
    const { getByText, getByLabelText } = renderTheme(
      <Header headerDashboard={true} token={true} />
    );

    expect(getByText('EasySchebule')).toBeInTheDocument();
    expect(getByLabelText('username')).toBeInTheDocument();
    expect(getByText('Sair')).toBeInTheDocument();
  });

  test('renders correctly without user menu', () => {
    const { getByText } = renderTheme(<Header headerDashboard token={false} />);

    expect(getByText('EasySchebule')).toBeInTheDocument();
    expect(getByText('Entrar')).toBeInTheDocument();
    expect(getByText('Criar conta')).toBeInTheDocument();
  });

  test('should match to snapshot', () => {
    const { container } = renderTheme(<Header headerDashboard token={false} />);

    expect(container).toMatchSnapshot();
  });
});
