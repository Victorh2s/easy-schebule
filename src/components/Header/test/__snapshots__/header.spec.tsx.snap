// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > should match to snapshot 1`] = `
.c0 {
  background: transparent;
  padding: 2.5rem;
}

.c1 {
  width: 100%;
  max-width: 1240px;
  margin: 0 auto;
  padding: 0 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 h2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 nav ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 li {
  list-style: none;
}

.c1 li button {
  background: transparent;
  border: none;
  color: #FFFFFF;
  font-weight: bold;
}

.c1 .user-menu {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-right: 1px solid #FFFFFF;
  padding: 0 1.5rem;
}

.c1 .user-menu a + span {
  margin-left: 1rem;
}

.c1 .user-menu a,
.c1 .user-menu span {
  display: inline-block;
  color: #FFFFFF;
  font-weight: bold;
}

.c1 li + li {
  margin-left: 2rem;
}

.c1 li + li button {
  cursor: pointer;
  color: #FFFFFF;
}

.c1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FFFFFF;
  font-weight: bold;
}

@media (max-width:700px) {
  .c1 {
    padding: 0 0.5rem;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c1 svg {
    margin-right: auto;
    cursor: pointer;
  }

  .c1 div {
    width: 100%;
    border: 1px solid black;
  }

  .c1 div h1 {
    font-size: 1rem;
  }

  .c1 div a {
    font-size: 1rem;
  }
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <h2>
        EasySchebule
      </h2>
      <nav>
        <ul>
          <li>
            <a
              href="/login"
            >
              Entrar
            </a>
          </li>
          <li>
            <a
              href="/registro"
            >
              Criar conta
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
   
</div>
`;
